stl/inc/valarray

```diff
@@ -1537,7 +1537,8 @@ public:
             _Ans += _Indexarr[_Idx] * _Stride[_Idx]; // compute offset
         }
 
-        while (0 < _Idx--) {
+        while (0 < _Idx) {
+            --_Idx;

```

Comment:
I'm not sure what's the motivation.
If the performance, then I don't think there's noticeable codegen difference.
If the clarity  then with `0 < _Idx` comparison and other stuff here it is still not much better

Reply:
On another thought, `0 < _Idx` -> `_Idx > 0` can be done in another change